x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportGetTransformStatsAction.java

```diff
@@ -131,17 +132,18 @@ protected void taskOperation(Request request, TransformTask task, ActionListener
 
     @Override
     protected void doExecute(Task task, Request request, ActionListener<Response> finalListener) {
+        final ClusterState state = clusterService.state();
+        TransformNodes.warnIfNoTransformNodes(state);
+
         transformConfigManager.expandTransformIds(
             request.getId(),
             request.getPageParams(),
             request.isAllowNoMatch(),
             ActionListener.wrap(hitsAndIds -> {
                 request.setExpandedIds(hitsAndIds.v2());
-                final ClusterState state = clusterService.state();
-                TransformNodeAssignments transformNodeAssignments = TransformNodes.transformTaskNodes(hitsAndIds.v2(), state);
-                // TODO: if empty the request is send to all nodes(benign but superfluous)
-                request.setNodes(transformNodeAssignments.getExecutorNodes().toArray(new String[0]));
-                super.doExecute(task, request, ActionListener.wrap(response -> {
+                final TransformNodeAssignments transformNodeAssignments = TransformNodes.transformTaskNodes(hitsAndIds.v2(), state);
```

Comment:
Is it possible that the `state` taken at line 135 is outdated here? If so, is it a problem? I guess not because the node role cannot change via cluster state update but wanted to double-check.

Reply:
In theory another request could come in and a transform gets started or stopped while stats is running. I think this issue is benign, worst case you get outdated stats or a wrong state. This can happen already today.